### DISCUSSION TITLE
TT-Metal Licensing and new contribution requirements

### DISCUSSION THREAD

**Discussion by tt-rkim** [tenstorrent/tt-metal#2465] [Contributor announcements]
cc: @tenstorrent-metal/developers @tenstorrent-metal/external-developers 

Hey everyone,

Per direction from and much thanks to Legal and @olofj, we now officially use the Apache-2.0 license in our repository.

What this means for us specifically is that new files must have an SPDX copyright statement and a license code at the top as a header. This includes sources AND test files AND documentation.

We have made changes to C, C++, and Python sources. More will be coming.

CONTRIBUTING.md has been updated to reflect this new style. Please read the change.

Please ensure we're all following these conventions.

For those who work with contractors / external devs, please ensure everyone understands these new requirements.

**Comment by tt-rkim** [comment]
cc: @pgkeller @TT-billteng @mo-tenstorrent 

For those working on other repositories, I would encourage similar changes to be planned and made as all of our stuff is going to be open-source eventually.

This includes sfpi, our fork of tracy (I think?), and UMD

**Reply by TT-billteng** [reply]
did you use a script to prepend the SPDX copyright header @tt-rkim ? We definitely need to do it for UMD

As for tracy the license is https://github.com/tenstorrent-metal/tracy/blob/master/LICENSE

Do we still add our license if it already has one?

**Reply by tt-rkim** [reply]
Then no - we do not. Thanks for pointing that out

And as for the script these are the commands for C/C++ and python specifically: https://gist.github.com/tt-rkim/92a031ece2f81ff876f4604db78c8e8f

However it's tt-metal specific so it excludes a lot of folders. You guys probably don't have to worry about that though

**Comment by tt-aho** [comment]
Do you think we should add this check for the headers to pre-commit hooks?

**Reply by tt-rkim** [reply]
I most definitely want that and as a `push: **` trigger for all branches.

Just need some time to do so 😵‍💫
